lib/project_types/script/layers/infrastructure/rust_project_creator.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class RustProjectCreator
+        include SmartProperties
+        property! :ctx, accepts: ShopifyCli::Context
+        property! :extension_point, accepts: Domain::ExtensionPoint
+        property! :script_name, accepts: String
+        property! :path_to_project, accepts: String
+
+        ORIGIN_BRANCH = 'main'
+        SAMPLE_PATH = 'default'
+
+        def setup_dependencies
+          git_init
+          setup_remote
+          setup_sparse_checkout
+        end
+
+        def bootstrap
+          pull
+          clean
```

Comment:
Just a heads up that the current flow is:
1. `project_creator.setup_dependencies`
2. `task_runner.install_dependencies`
3. `project_creator.bootstrap`

which means that for rust, installing via the task runner wouldn't work right now since the `Cargo.toml` gets created during this `bootstrap`. So it might make more sense to move `pull` and `clean` into `#setup_dependencies` for now?

If sparse checkout is the right longterm approach then hopefully we can simplify the flow to be `project_creator.bootstrap` then `task_runner.install_dependencies` for all languages 🤞

Reply:
> which means that for rust, installing via the task runner wouldn't work right now since the Cargo.toml gets created during this bootstrap. So it might make more sense to move pull and clean into #setup_dependencies for now?

Yeah, this part should be tackled in: https://github.com/Shopify/script-service/issues/2455. We should move everything under `setup_dependencies` and make `bootstrap` a noop. I guess we could do the same for the task runner's `install_dependencies`: make it noop for now and when pushing handle everything under the task runner's `build`.

Does that make sense?

> If sparse checkout is the right longterm approach then hopefully we can simplify the flow to be project_creator.bootstrap then task_runner.install_dependencies for all languages 🤞

Yeah, if this is the long term approach we can invert the order of operations in such a way that they are logically ordered, bootstrap -> install deps.

Reply:
Actually, I'm going to move everything under `setup_dependencies` in this PR already

Reply:
Sounds good!